### PR TITLE
Testing environment Config update

### PR DESCRIPTION
**Pull request by karthikmsqa**
# Description

This PR includes the following proposed change(s):

- Documentation update
- Maven profiles for different environments
- Simplified config properties files
- Modified cucumber tests workflow

## Type of change

- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -41,9 +54,9 @@ Running the tests create an html report [here](test-output/extent/HtmlReport/Ext
 
 We support running the tests with the current tags:
 
-### @backend
+### Tags
 
-This would only run backend tests.
+Using tags would only run the specific tests.
```
Comment (sdevalapurkar) [inline tests/README.md:59]:
```suggestion
Using tags will only run the specified tests.
```

**Review by sdevalapurkar — APPROVED** [review]
LGTM, few suggestions

File: tests/README.md (hunk 1 above)
Comment (karthikmsqa) [reply tests/README.md]: Thanks @sdevalapurkar

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java [hunk 2]
```diff
@@ -4,11 +4,15 @@
 
     private String validExistingCSOGuid;
     private String invalidNoFilingRoleGuid;
-    private String nonExistingCSOGuid;
+
```
Comment (akroon3r) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java:7]:
```suggestion
```

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java [hunk 3]
```diff
@@ -4,11 +4,15 @@
...
     private String validUserId;
     private String inValidUserId;
     private String clientSecret;
 
+    public CsoAccountGuid() {
```
Comment (akroon3r) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java:12]: If it's empty, why do we need it?

**Review by akroon3r — COMMENTED** [review]
Couple of questions.

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java (hunk 3 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/testdatatypes/CsoAccountGuid.java:8]: Removed. Thanks @akroon3r

**Review by akroon3r — APPROVED** [review]

**Pull request merged**